examples/js/archive.js

```diff
@@ -300,11 +302,88 @@ saveMap.addEventListener('click', () => {
       a.download = fileName ? fileName + '.json' : 'MapknitterLite.json';
       a.click();
     }
-})
+});
 
 // share map modal
 const shareModal = document.getElementById('shareModal')
 const modality =  new bootstrap.Modal(shareModal)
 shareMapBtn.addEventListener('click', () => {
   bootstrap.Modal.getInstance(shareModal).show()
-})
\ No newline at end of file
+});
+
+// -----------------------------------------------------------------------------
+function extractFileName(name) {
+  const startIndex = name.lastIndexOf('.');
+  const fileName = name.substring(0, startIndex);
+
+  return fileName;
+}
+
+function handleDrop (e) {
+  const files = e.dataTransfer.files;
+  const reader = new FileReader();
+  
+  // confirm file being dragged has json format
+  if (files.length === 1 && files[0].type === 'application/json') { 
+    reader.addEventListener('load', () => {
+      let imgUrl;
+      let options;
+      const imgObj = JSON.parse(reader.result);
+      // for json file with multiple image property sets
+      if (imgObj.images.length > 1) {
```

Comment:
Here, should we be reading `.images`? When I look at an example JSON file that's been generated from MapKnitter lite, it doesn't have this property, it's just an array, like this:

```json
[{"id":1504,"src":"https://archive.org/download/sanborn-1885/sanborn-1885.jpg","width":7573,"height":6396,"image_file_name":"sanborn-1885.jpg","nodes":[{"lat":39.32869582897733,"lon":-120.187548995018},{"lat":39.32982863334289,"lon":-120.18338084220888},{"lat":39.3270941035765,"lon":-120.182141661644},{"lat":39.32596125491416,"lon":-120.18630981445314}],"cm_per_pixel":10.875006010132818}]
```

Reply:
Oh! okay.. I will look into this.

Reply:
Hi @jywarren, so I tested the feature with `segeotest2.json` and everything works fine, so no problem really. However, if you compare this file with `test2.json` (generated from Mapknitter lite, that you used in your test), you will observe a difference. Some properties (e.g., `avg_cm_per_pixel`, `images` - this is an array) are missing `in test2.json` but available in `segeotest2.json`. That's why the test you did failed.

I think format in `segeotest2.json` correctly matches the expected JSON output from Illustration 7 and that there's instead a need to rework Mapknitter Lite to produce JSON output similar to `segeotest2.json` . **What do you think?**  

See illustrations 5, 6 & 7 below.

**ILLUSTRATION 5 - test2.json (generated from Mapknitter Lite)**
![test2](https://user-images.githubusercontent.com/1612359/217041386-7f5826e6-a0f6-4b3b-980f-4afeee3ee619.PNG)

**ILLUSTRATION 6 - segeotest2.json**
![segeotest2](https://user-images.githubusercontent.com/1612359/217038378-4b708154-265c-4583-a1e9-041a6ed4d256.png)

**ILLUSTRATION 7 - code snippet**
![illustration7](https://user-images.githubusercontent.com/1612359/217044862-ff73149f-c5dd-417b-a096-c52b302a1ca6.PNG)

Reply:
Hi, thanks for figuring this out! Where is your JSON file from? It looks like MK Lite follows the format shown here: 

https://github.com/publiclab/image-sequencer-app

Of course we could accommodate both formats. But it does seem helpful to know where the two each come from and to standardize. One reason to use the version listed at the URL above is that there is a pregenerated saved JSON file for every single map in the old MapKnitter application, in the internet archive:

https://web.archive.org/web/20220726175911/https://mapknitter.org/maps/ceres--2/warpables.json

So in theory we could adapt this to display any legacy MapKnitter map, if we maintain the same exact format.

Reply:
Okay, as for where the two formats came from:
1. test2.json: if you click save icon on MK lite, you get a json file formatted into what obtains in test2.json.
2. segeotest2.json: this comes from the json object returned from `generateExportJson()` in distortableCollection.js. 

Note that the format shown in https://github.com/publiclab/image-sequencer-app is at variance with the json object returned by  `generateExportJson()` .

**So should we support both formats?**

Reply:
Gosh, this is a mess, sorry! I looked carefully and actually image-sequencer-app accepts an object with a `collection` array, not an `images` array; you can see it here:

https://github.com/publiclab/Leaflet.DistortableImage/blob/8cca752ca55b4aaa5ccedcb2689589e0f2eab1ff/examples/js/export.js#L51

and here:

https://github.com/publiclab/image-sequencer-app#usage

and unfortunately, we even convert from `images` to `collection` here:

https://github.com/publiclab/Leaflet.DistortableImage/blob/8cca752ca55b4aaa5ccedcb2689589e0f2eab1ff/src/edit/DistortableCollection.Edit.js#L327-L330

This must have happened as multiple people over the years wrote code and didn't manage to coordinate. It really should be standardized. I believe `generateExportJson` is the only place `images` appears; elsewhere, I believe we use `collection`. Let's stick with that, what do you think:

```
{
  "collection": [
    { "nodes": [ ... ], src: "" },
    { "nodes": [ ... ], src: "" }
  ]
}
```

Reply:
I'm almost tempted to try to fix this internal inconsistency, as it's annoying and confusing. But for now, maybe, let's just get the output and repackage it using `collection` as in the `DistortableCollection.Edit.js` code above. How does that sound?

Reply:
> I'm almost tempted to try to fix this internal inconsistency, as it's annoying and confusing. But for now, maybe, let's just get the output and repackage it using `collection` as in the `DistortableCollection.Edit.js` code above. How does that sound?

Since `images` is only used in `generateExportJson`, your suggestion sounds good. 

However, on this strength and if benign, probably an alternative approach could be to refactor `generateExportJson` to return an object formatted into `collection` as presented in code snippet above. **This way, we bite the bullet and make the story straight at once**. In any case, other dependent source files (I believe recent files only) will also be refactored accordingly. What do you think?